### ISSUE TITLE
add user with user module for RHEL7 server failed

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29737]
From @FuYuanZhao on 2016-07-22T13:44:35Z

##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

user
##### ANSIBLE VERSION

```
ansible: ansible-2.1.0.0-1.el6.noarch
```
##### SUMMARY

I have a playbook to create a local user, but when I ran this playbook for two servers, on is RHEL6, the other is RHEL7, why failed on RHEL7, show me the following:

```
 fatal: [rht04]: FAILED! => {"changed": false, "failed": true, "msg": "usermod: user 'ansible' does not exist\n", "name": "ansible", "rc": 6}
•name: Add ansible local user to the server user: name: ansible uid: 600 comment: "Ansible admin user" home: /opt/staff/ansible createhome: yes state: present

TASK [ansiblelocaluser : Add ansible local user to the server] *****************
 ok: [rht02] => {"append": false, "changed": false, "comment": "Ansible admin user", "group": 600, "home": "/opt/staff/ansible", "move_home": false, "name": "ansible", "shell": "/bin/bash", "state": "present", "uid": 600}
 fatal: [rht04]: FAILED! => {"changed": false, "failed": true, "msg": "usermod: user 'ansible' does not exist\n", "name": "ansible", "rc": 6}
 TASK [ansiblelocaluser : set up authorized_key for ansible local user] *********
 ok: [rht02] => {"changed": false, "exclusive": false, "key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCyAACMvkERvxgdrthtuK1gOmj88FV+Hr+gmvlMr13Kv+It+kqXF7KgfTzwD5RuZcABU30Hw+IzTe5D2IQQFx4vGOxfGPGhiadfelktLPXj2KaZSYVlfkEWtJ55VG5sEOKx8tMpMXjOMchggstTFe6+/5zCdSeb+29sAUulWHR+pom/eAmHuwO1PiW23hHWiQULzzDxwWVDnqTLvXC1qtjE4OFjmCnsiKTK7AuuU9mqvhS1bcw20hZ8o3lbkgXk0Sr9prwO39B5H8JjVwVssW2l+CqWs+qwGNOSk4Y1fTkrwAj6ZksWGs4br0+Nnyji+Y/V0NkrhVGZh0ybWkF2XFN9 ansible@ansible01", "key_options": null, "keyfile": "/opt/staff/ansible/.ssh/authorized_keys", "manage_dir": true, "path": null, "state": "present", "unique": false, "user": "ansible", "validate_certs": true}
 TASK [ansiblelocaluser : make sure that ansible user are in the Unix_ADM group in /etc/sudoers] ***
 ok: [rht02] => {"backup": "", "changed": false, "msg": ""}
 NO MORE HOSTS LEFT *************************************************************
 to retry, use: --limit @configure_ansible_env.retry
 PLAY RECAP *********************************************************************
 rht02 : ok=5 changed=0 unreachable=0 failed=0 
 rht04 : ok=2 changed=1 unreachable=0 failed=1
```

**NOTE: AD and Centrify Direct Control as user authentication**
##### STEPS TO REPRODUCE

```

```

##### ACTUAL RESULTS

```

```

Copied from original issue: ansible/ansible-modules-core#4238

**Comment by ansibot** [comment]
From @jctanner on 2016-07-22T13:44:35Z

@FuYuanZhao please provide the "ansible version" output.

**Comment by ansibot** [comment]
From @gamuniz on 2016-07-22T13:44:35Z

ansible: ansible-2.1.0.0-1.el6.noarch

**Comment by ansibot** [comment]
From @gamuniz on 2016-07-22T13:44:35Z

```
# /etc/nsswitch.conf
#
# An example Name Service Switch config file. This file should be
# sorted with the most-used services at the beginning.
#
# The entry '[NOTFOUND=return]' means that the search for an
# entry should stop if the search in the previous entry turned
# up nothing. Note that if the search failed due to some other reason
# (like no NIS server responding) then the search continues with the
# next entry.
#
# Valid entries include:
#
# nisplus Use NIS+ (NIS version 3)
# nis Use NIS (NIS version 2), also called YP
# dns Use DNS (Domain Name Service)
# files Use the local files
# db Use the local database (.db) files
# compat Use NIS on compat mode
# hesiod Use Hesiod for user lookups
# [NOTFOUND=return] Stop searching if not found so far
#
# To use db, put the "db" in front of "files" for entries you want to be
# looked up first in the databases
#
# Example:
#passwd: db files nisplus nis
#shadow: db files nisplus nis
#group: db files nisplus nis
passwd: centrifydc files
shadow: centrifydc files
group: centrifydc files
#hosts: db files nisplus nis dns
hosts: files dns
# Example - obey only what nisplus tells us...
#services: nisplus [NOTFOUND=return] files
#networks: nisplus [NOTFOUND=return] files
#protocols: nisplus [NOTFOUND=return] files
#rpc: nisplus [NOTFOUND=return] files
#ethers: nisplus [NOTFOUND=return] files
#netmasks: nisplus [NOTFOUND=return] files
bootparams: nisplus [NOTFOUND=return] files
ethers: files
netmasks: files
networks: files
protocols: files
rpc: files
services: files
netgroup: nisplus
publickey: nisplus
automount: files nisplus
aliases: files nisplus 
```

**Comment by ansibot** [comment]
From @bcoca on 2016-07-22T13:44:35Z

does the ansible user already exist in centrify?

**Comment by ansibot** [comment]
From @FuYuanZhao on 2016-07-22T13:44:35Z

Yes , but when I create a local user "ansible" on the targeted server ,I will add ansible user to the user.ignore file, that's the same way I did on RHEL6 servers

**Comment by ansibot** [comment]
From @jctanner on 2016-07-22T13:44:35Z

ping @grossws just letting you know this exists.

**Comment by ansibot** [comment]
From @jctanner on 2016-07-22T13:44:35Z

@FuYuanZhao let us know what you find out, thanks.

needs_info

**Comment by ansibot** [comment]
From @buggsdummy on 2016-07-22T13:44:35Z

TASK [add jenkins user] ********************************************************
fatal: [x.x.x.x]: FAILED! => {"changed": false, "failed": true, "msg": "usermod: user 'jenkins' does not exist in /etc/passwd\n", "name": "jenkins", "rc": 6}

Along with that, maybe have ansible check with id <user> and if it returns a user then go instead of looking in /etc/passwd.

**Comment by ansibot** [comment]
From @ansibot on 2016-07-22T13:44:35Z

@FuYuanZhao ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @ansibot on 2016-07-22T13:44:35Z

@FuYuanZhao ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @sfromm
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @ansibot on 2016-07-22T13:44:35Z

@FuYuanZhao ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@FuYuanZhao This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@FuYuanZhao You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
